### PR TITLE
Complete rewrite to functional programming

### PR DESCRIPTION
**Pull request by alanmarcell**

### CODE REVIEW

**Review by angeloocana — COMMENTED** [review]
I'm thinking if BaseApp needs to die... What do you think?

File: package.json [hunk 1]
```diff
@@ -33,19 +35,17 @@
...
-    "@types/node": "^7.0.12",
-    "opn-cli": "^3.1.0"
+    "sinon": "^2.3.5",
+    "tslint": "^5.4.3",
+    "typescript": "^2.3.4"
   },
   "dependencies": {
-    "ptz-log": "^1.0.0",
-    "ptz-core-domain": "^1.2.0"
+    "@alanmarcell/ptz-core-domain": "^0.0.1",
```
Comment (alanmarcell) [inline package.json:47]: I'm using this scoped version until we have a stable version

File: src/BaseApp.ts [hunk 2]
```diff
@@ -1,15 +1,11 @@
...
-    }
-
-    log(...args: any[]): void {
-        if (this.logFunc)
-            this.logFunc(...args);
-    }
-}
+import { IBaseApp, IBaseAppArgs } from '@alanmarcell/ptz-core-domain';
+
+export const createBaseApp = (baseAppArgs: IBaseAppArgs): IBaseApp => {
```
Comment (alanmarcell) [inline src/BaseApp.ts:3]: It's not doing too much. Shall it die?

Comment (angeloocana) [reply src/BaseApp.ts:3]: Die OO!!!

**Pull request merged**